homeassistant/components/mqtt/alarm_control_panel.py

```diff
@@ -178,9 +177,7 @@ def _setup_from_config(self, config: ConfigType) -> None:
 
         if (code := self._config.get(CONF_CODE)) is None:
             self._attr_code_format = None
-        elif code == REMOTE_CODE or (
-            isinstance(code, str) and re.search("^\\d+$", code)
-        ):
+        elif code == REMOTE_CODE or (isinstance(code, str) and code.isdigit()):
```

Comment:
```suggestion
        elif code == REMOTE_CODE or str(code).isdigit()):
```

Theoretical case can be `bytes` or str. In this case we can just add a `str` wrapper, that costs less I think.

Reply:
That would be a functional "while at it" change in a refactoring PR though and the way I see it, as such, against the "perfect PR" as documented: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

I don't mind that much, but I'd rather leave such changes to people actually knowing/working on affected components.

Reply:
In this case, I know the code, but you are right it would be better to split this to a different PR.